src/components/Navbar/SettingsDrawer/ReciterSelectionBody.tsx

```diff
@@ -70,34 +71,39 @@ const SettingsReciter = () => {
             ? filterReciters(data.reciters, searchQuery)
             : data.reciters;
           return (
-            <div>
-              {filteredReciters
+            <RadioGroup
+              value={selectedReciter.id.toString()}
+              orientation={RadioGroupOrientation.Vertical}
+              label="reciter"
+              onChange={(newId) => onSelectedReciterChange(newId, data.reciters)}
+              items={filteredReciters
                 .sort((a, b) => (a.name > b.name ? 1 : -1))
-                .map((reciter) => (
-                  <label
-                    className={styles.reciter}
-                    htmlFor={reciter.id.toString()}
-                    key={reciter.id}
-                  >
-                    <input
-                      id={reciter.id.toString()}
-                      type="radio"
-                      name="reciter"
-                      value={reciter.id}
-                      checked={reciter.id === selectedReciter.id}
-                      onChange={(e) => {
-                        onSelectedReciterChange(e.target.value, data.reciters);
-                      }}
-                    />
-                    <span lang={reciter.translatedName.languageName}>
-                      {reciter.translatedName.name}
-                    </span>
-                    {reciter.style.name !== DEFAULT_RECITATION_STYLE && (
-                      <span className={styles.recitationStyle}>{reciter.style.name}</span>
-                    )}
-                  </label>
-                ))}
-            </div>
+                .map((reciter) => ({
+                  value: reciter.id.toString(),
+                  id: reciter.id.toString(),
+                  label:
+                    reciter.style.name !== DEFAULT_RECITATION_STYLE
+                      ? `${reciter.translatedName.name}/${reciter.style.name}`
+                      : reciter.translatedName.name,
```

Comment:
hmm, i see. So what we're doing here is basically we put the 2 informations inside a string and we seperate it with a  `/` right.

I know the current dls doesn't support rendering the recitation style in a radio component. 

Would it be better to expose the RadioGroup root component as well? (now that we already expose `.Item` and `.Indicator`)

So we can do 

```
<RadioGroup.Root>
   {filteredReciters.map( reciter =>
     <>
        <RadioGroup.Item ... />
        <label>...</label>
     </>
    )
</RadioGroup.Root>
```

This way we don't need to 
1. Encode 2 informations inside a string inside `label`
2. introduce renderProps `renderItem`

it's much more composable.

Reply:
We can use `RadioGroup` for simpler use case.

and use the `Root` `Item` and `Indicator` for a complex use case that needs a lower level API

Reply:
Salam bro @muhajirdev, yeah makes sense sorry I must've wrote it before sleep 😄

Reply:
no worries, may Allah reward you for the contributions 😁